scalding-beam/src/main/scala/com/twitter/scalding/beam_backend/BeamBackend.scala

```diff
@@ -61,8 +74,10 @@ object BeamPlanner {
           BeamOp.Source(config, src, srcs(src))
         case (IterablePipe(iterable), _) =>
           BeamOp.FromIterable(iterable, kryoCoder)
-        case (wd: WithDescriptionTypedPipe[a], rec) =>
-          rec[a](wd.input)
+        case (wd: WithDescriptionTypedPipe[_], rec) => {
+          val op = rec(wd.input)
+          op.withName(wd.descriptions.map(_._1).head)
```

Comment:
`.head` can throw. Can we instead do:
```scala
wd.descriptions match {
  case head :: _ =>
    op.withName(head._1)
  case Nil =>
    op
}
```

Reply:
Done. I thought that for a `WithDescriptionTypedPipe` this could not be empty.

Reply:
it maybe can't but we didn't make that clear in the types... :/